cmd/compute-domain-controller/mnsdaemonset.go

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2025 NVIDIA CORPORATION.  All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+
+	nvapi "github.com/NVIDIA/k8s-dra-driver-gpu/api/nvidia.com/resource/v1beta1"
+)
+
+// MultiNamespaceDaemonSetManager manages DaemonSets across multiple namespaces.
+type MultiNamespaceDaemonSetManager struct {
+	config   *ManagerConfig
+	managers map[string]*DaemonSetManager
+}
+
+// NewMultiNamespaceDaemonSetManager creates a new multi-namespace DaemonSet manager
+// It creates individual DaemonSet managers for the driver namespace and all additional namespaces.
+func NewMultiNamespaceDaemonSetManager(config *ManagerConfig, getComputeDomain GetComputeDomainFunc) *MultiNamespaceDaemonSetManager {
+	m := &MultiNamespaceDaemonSetManager{
+		config:   config,
+		managers: make(map[string]*DaemonSetManager),
+	}
+
+	// Use a map to deduplicate namespaces (driver namespace + additional namespaces)
+	uniqueNamespaces := make(map[string]bool)
```

Comment:
nit: you could also use `map[string]struct{}` to represent your set. `struct{}` takes zero memory

From an [SO answer](https://stackoverflow.com/a/37320521):

> The struct type with no fields is called the empty struct, written struct{}. It has size zero and carries no information but may be useful nonetheless. Some Go programmers use it instead of bool as the value type of a map that represents a set, to emphasize that only the keys are significant, but the space saving is marginal and the syntax more cumbersome, so we generally avoid it.


```suggestion
	uniqueNamespaces := make(map[string]struct{})
```

Reply:
This is how I used to do it, but @elezar argued it was more readable with bools and I don't feel strongly, so we standardized on using bools when we need constructs like this.

Reply:
I went with `uniqueNamespaces := make(map[string]struct{})` in the end.